python-service/src/auth_validator.py

```
@@ -0,0 +1,177 @@
+
+    The manifest describes which snapshots to restore and to which
+    destination paths.
+
+    VULN-7 (Insecure deserialization / arbitrary code execution):
+    ``yaml.load`` with ``Loader=yaml.FullLoader`` (the default before
+    PyYAML 5.1) or any Loader other than ``SafeLoader`` can execute
+    arbitrary Python via ``!!python/object/apply`` tags in the YAML.
+    Even ``FullLoader`` is exploitable in some PyYAML versions.
+    ``yaml.safe_load`` should be used here.
+    """
+    return yaml.load(manifest_yaml, Loader=yaml.FullLoader)
```

Ruff (0.15.15)

[error] Probable use of unsafe loader `FullLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)